contracts/l1/OptimismResolver.sol

```diff
@@ -39,19 +54,31 @@ contract OptimismResolver is IExtendedResolver, SupportsInterface, OwnedENSNode
         emit NewOwner(owner);
     }
 
-    //This function only exists during develoopent. Will be removed before releasing prod
+    function setL2Resolver(address _newL2Resolver) external onlyOwner {
+        l2Resolver = _newL2Resolver;
+        emit NewOwner(owner);
+    }
+
     function setUrl(string memory _url) external onlyOwner {
         url = _url;
     }
 
+    function setGraphUl(string memory _graphqlUrl) external onlyOwner {
+        graphqlUrl = _graphqlUrl;
+    }
+
     /**
      * Resolves a name, as specified by ENSIP 10.
      * @param name The DNS-encoded name to resolve.
      * @param data The ABI encoded data for the underlying resolution function (Eg, addr(bytes32), text(bytes32,string), etc).
      * @return The return data, ABI encoded identically to the underlying function.
      */
     function resolve(bytes calldata name, bytes calldata data) external view override returns (bytes memory) {
-        bytes memory callData = abi.encodeWithSelector(IResolverService.resolve.selector, name, replaceNodeWithOwnedNode(data));
+        bytes32 node = bytes32(data[4:36]);
+        address nodeOwner = ensRegistry.owner(node);
```

Comment:
This is fine for non wrapped contract but you have to call `NameWrapper.ownerOf()` for wrapped one. Please see "Owner" section of https://enslabs.notion.site/NameWrapper-third-party-support-guide-06b45e801830414da9825c8ffb747a30 for more detail